lib/SILOptimizer/Transforms/TempRValueElimination.cpp

```diff
@@ -628,6 +628,16 @@ TempRValueOptPass::tryOptimizeStoreIntoTemp(StoreInst *si) {
     // We pass in SILValue() since we do not have a source address.
     if (!collectLoads(useOper, user, tempObj, SILValue(), loadInsts))
       return {std::next(si->getIterator()), false};
+
+    // Bail if there is any kind of user which is not handled in the code below.
+    switch (user->getKind()) {
+      case SILInstructionKind::CopyAddrInst:
+      case SILInstructionKind::FixLifetimeInst:
+      case SILInstructionKind::MarkDependenceInst:
+        continue;
+      default:
+        return {std::next(si->getIterator()), false};
```

Comment:
I want to think about this a little more, but can you change this to crash in asserts builds? I think that might be a nice way to square the circle of not crashing in non-asserts builds but also getting noise when we do not update all of the relevant code. Creating implicit dependencies in the code like this, makes me nervous.

Reply:
No, that doesn't make sense.
Why should there be an assert for a case which totally legal to happen and we know how to reproduce it?

Reply:
Ah ok. I understand what you are doing now. LGTM.